core/src/ModelArray.cpp

```diff
@@ -179,22 +177,29 @@ void ModelArray::setData(double value)
 void ModelArray::setData(const double* pData)
```

Comment:
The class member `m_data` is defined to use doubles, and it should be possible to feed a (correctly sized) double buffer to the array. This is tested in ModelArray_test.cpp line 71. It's possible that this could be refactored, but that should be a new issue.